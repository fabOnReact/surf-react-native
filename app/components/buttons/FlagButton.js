import Icon from 'react-native-vector-icons/Ionicons';

export default function FlagButton({ setVideo }){
  return (
    <TouchableOpacity
      text="Submit"
      style={styles.containerRight}
      onPress={() => setVideo(false)} >
      <Text>Cancel</Text>
      <Icon name="ios-flag" 
        size={50} 
        color="black" />
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  containerRight: {
    position: "absolute",
    bottom: 120,
    right: 5,
    height: 50,
    width: 100,
    backgroundColor: 'white',
    borderRadius: 15,
    zIndex: 1,
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-evenly',
  },
})
