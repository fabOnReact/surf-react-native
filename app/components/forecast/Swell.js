import React, { Component } from 'react';
import { StyleSheet, Text } from 'react-native';
import Unit from '../../lib/unit';

export default class Swell extends Component {
  render() {
    const { text, styles} = this.props
    return (
      <Text style={styles}>
        { text }
      </Text>
    )
  }
}
