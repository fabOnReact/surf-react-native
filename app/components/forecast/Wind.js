import React, { Component } from 'react';
import { Text, View, Image } from 'react-native';
import { H3 } from 'native-base';
import { header } from './styles';

export default class Wind extends Component {
  render() {
    const { windSpeed, styles } = this.props
    return (
      <React.Fragment>
        <Text style={styles}>
          { windSpeed } m/s wind
        </Text>
      </React.Fragment>
    )
  }
}
