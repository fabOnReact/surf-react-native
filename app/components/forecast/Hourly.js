import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import Swell from '../forecast/Swell';
import Wind from '../forecast/Wind';
import Arrow from '../icons/Arrow';
import DeviceInfo from 'react-native-device-info';
import { header } from '../forecast/styles';

export default class Hourly extends Component {
  renderHourly(hourly) {
    const { 
      waveHeight, swellHeight, swellPeriod, 
      optimal_swell, swellDirection, swellDirectionInWord, 
      windSpeed, windDirection, optimal_wind, windDirectionInWord,
    } = hourly
    const swellIcon = require('../../images/down-cursor-black.png')
    const windIcon = require('../../images/down-arrow-black.png')
    return (
      <React.Fragment>
        <View style={styles.container}>
          <Arrow
            offshore={optimal_swell}
            direction={swellDirection}
            icon={swellIcon} />
          <Swell
            period={swellPeriod} 
            swellHeight={swellHeight} 
            styles={[
              { marginLeft: 10 },
              header.shadowHeader,
            ]}
          />
          <Wind 
            windSpeed={windSpeed}
            styles={[
              { marginLeft: 10 },
              header.shadowHeader,
            ]}
          />
          <Arrow
            offshore={optimal_wind}
            direction={windDirection}
            icon={windIcon}
          />
        </View>
      </React.Fragment>
    )
  }

  render() {
    const {hourly } = this.props
    const { swellHeight } = hourly
    return (
      <React.Fragment>
        { !!swellHeight && this.renderHourly(hourly) }
      </React.Fragment>
    )
  }
}

const has_notch = DeviceInfo.hasNotch()
export const styles = StyleSheet.create({
  container: {
    width: "100%",
    position: 'absolute',
    top: 10,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    padding: 15,
  }, 
})

      {/*
      <View style={{position: "absolute", top: 0}}>
        <View style={styles.flexbox}>
          <Image 
            source={require('../../images/down-cursor-black.png')}
            style={[
              { left: "15%", tintColor: swellColor },
              styles.icon, 
              { transform: [{ rotateZ: `${swellDirection}deg`}] } 
            ]}
          />
          <Image 
            source={require('../../images/down-arrow-black.png')}
            style={[ 
                { right: "15%", tintColor: windColor },
                styles.icon,
                { transform: [{ rotateZ: `${windDirection}deg`}] }
            ]}
          />
        </View>
        <Text style={[styles.shadowHeader, { position: "absolute", top: "87%", left: "25%" }]}>
          { swellDirectionInWord }
        </Text>
        <Text style={[styles.shadowHeader, { position: "absolute", top: "87%", right: "25%" }]}>
          { windDirectionInWord }
        </Text>
        <Text style={[styles.shadowHeader, { position: "absolute", top: "93%", left: "5%" }]}>
          @ { swellPeriod } seconds
        </Text>
        <Text style={[styles.shadowHeader, { position: "absolute", top: "93%", right: "5%" }]}>
          { optimal_wind != null && optimal_wind ? "offshore wind" : "onshore wind"  }
        </Text>
      </View>
      */}
