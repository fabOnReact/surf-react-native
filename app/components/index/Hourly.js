import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import Swell from '../forecast/Swell';
import Wind from '../forecast/Wind';
import Arrow from '../icons/Arrow';
import MenuButton from '../buttons/MenuButton';
import FlagButton from '../buttons/FlagButton';
import { header } from '../forecast/styles';
import Api from '../../lib/api';
import Data from '../../lib/data';

export default class Hourly extends Component {
  constructor(props) {
    super(props)
    this.state = { visible: false, locations: [] }
    this.api = new Api()
  }

  componentDidMount = async () => {
    this.api.page = 1
    this.api.per_page = 15
    const locations_request =  await this.api.getLocations({ flags: ["&with_cameras=true"] })
    // console.warn(locations_request);
    const locations = await locations_request.json()
    this.setState({ locations })
  }

  renderMenu = () => {
    const { navigation, imperial } = this.props
    const { locations } = this.state
    navigation.navigate("Menu", { imperial, locations })
  }

  onFlagPress = () => {
    const { navigation, post } = this.props
    navigation.navigate("Flag", { post: post })
  }

  renderHourly(data) {
    return (
      <React.Fragment>
        <Swell
          text={data.swell}
          styles={[
            header.text,
            header.shadowHeader,
          ]}
        />
        <Wind 
          text={data.wind}
          styles={[
            header.text,
            header.shadowHeader, 
          ]}
        />
      </React.Fragment>
    )
  }

  render() {
    const { location, post, imperial } = this.props
    const { visible, locations } = this.state
    const with_locations = locations.length > 0
    const { reported } = post
    const iconColor = reported ? "red" : "white"
    const { name, forecast_info: { hourly, tide_data }} = location
    const { swellHeight } = hourly
    const data = new Data({...hourly, imperial })

    return (
      <React.Fragment>
        <View style={styles.container}>
          <MenuButton 
            action={this.renderMenu}
          />
          <FlagButton 
            action={this.onFlagPress} 
            style={iconColor}
          />
          <Text style={[
            header.shadowHeader,
            header.text,
          ]}>
            { name }
          </Text>
          { !!swellHeight && this.renderHourly(data) }
        </View>
      </React.Fragment>
    )
  }
}

const has_notch = DeviceInfo.hasNotch()
export const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: has_notch ? 0 : 30,
    width: "100%",
    zIndex: 4,
    height: 90,
  }, 
})
