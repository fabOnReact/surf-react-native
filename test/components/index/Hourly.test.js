import React, { Component } from 'react';
import {shallow} from 'enzyme';
import Hourly from '../../../app/components/index/Hourly';
import DeviceInfo from 'react-native-device-info';
import FlagButton from '../../../app/components/buttons/FlagButton';
import Swell from '../../../app/components/forecast/Swell';
import Wind from '../../../app/components/forecast/Wind';
import Data from '../../../app/lib/data';
import Unit from '../../../app/lib/unit';

jest.mock('react-native-device-info', () => ({
  hasNotch: jest.fn()
}))

const location = { name: 'test', forecast_info: { hourly: [], tide_data: [] }}
const post = { 
  forecast: 
    { 
      "windSpeed": 3.5, 
      "waveHeight": 1, 
      "swellHeight": 1, 
      "swellPeriod": 13, 
      "optimal_wind": true, 
      "optimal_swell": true, 
      "waveDirection": 204, 
      "windDirection": 221, 
      "swellDirection": 204
    }
}

describe('Hourly', () => {
    describe('Rendering', () => {
        it('matches to snapshot', () => {
          const component = shallow(<Hourly />)
          expect(component).toMatchSnapshot()
        });

        it('renders the FlagButton component', () => {
          const component = shallow(<Hourly />)
          expect(component.find(FlagButton).length).toBe(1)
        });

        it('renders the Swell component', () => {
          jest.mock('../../../app/lib/data')
          jest.mock('../../../app/lib/unit')
          const unit = new Unit()
          console.log('unit', unit);
          const props = { location: { forecast_info: { hourly: {}}}}
          const component = shallow(<Hourly {...props} />)
          expect(component.find(Swell).length).toBe(1)
        });
    });
});

