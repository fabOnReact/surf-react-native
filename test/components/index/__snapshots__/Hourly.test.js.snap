// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hourly Rendering matches to snapshot 1`] = `
<Fragment>
  <View
    style={
      Object {
        "height": 90,
        "position": "absolute",
        "top": 30,
        "width": "100%",
        "zIndex": 4,
      }
    }
  >
    <FlagButton
      action={[Function]}
      style="white"
    />
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 5,
            "textShadowColor": "rgb(0, 0, 0)",
            "textShadowOffset": Object {
              "height": 2,
              "width": -2,
            },
            "textShadowRadius": 1,
          },
          Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginBottom": 5,
            "textAlign": "center",
            "textShadowColor": "rgb(0, 0, 0)",
            "textShadowOffset": Object {
              "height": 2,
              "width": -2,
            },
            "textShadowRadius": 1,
            "width": "100%",
          },
        ]
      }
      testID="postHeader"
    />
  </View>
</Fragment>
`;
